decaffeinate.sh

```diff
@@ -0,0 +1,45 @@
+set -ex
+
+npx bulk-decaffeinate convert --dir app/coffee
+
+npx bulk-decaffeinate clean
+
+git mv app/coffee app/js
+
+git commit -m "Rename app/coffee dir to app/js"
+
+npx prettier-eslint 'app/js/**/*.js' --write
+
+git add .
+git commit -m "Prettier: convert app/js decaffeinated files to Prettier format"
+
+npx bulk-decaffeinate convert --dir test/acceptance/coffee
+
+npx bulk-decaffeinate clean
+
+git mv test/acceptance/coffee test/acceptance/js
+
+git commit -m "Rename test/acceptance/coffee to test/acceptance/js"
+
+npx prettier-eslint 'test/acceptance/js/**/*.js' --write
+
+git add .
+git commit -m "Prettier: convert test/acceptance decaffeinated files to Prettier format"
+
+git mv app.coffee app.js
+git mv Gruntfile.coffe Gruntfile.js
+git mv config/settings.defaults.coffee config/settings.defaults.js
+
+git commit -m "Rename individual coffee files to js files"
+
+decaffeinate app.js
```

Comment:
Personally I would do the decaffeination before renaming the file (unless this causes a problem?)

Reply:
I've renamed first so that we preserve git history of individual files without having to go find the previous coffee file.

If we decaffeinate first, it also changes the filename to `*.js` and because the file is substantially changed, `git` doesn't recognize it as a rename and hence just thinks it's a new file and a deleted file.

I'm also pretty sure this is what `bulk-decaffeinate` does also, no? Looking at the commits in https://github.com/sharelatex/web-sharelatex-internal/pull/1037 it looks like it does a re-naming, then conversion. e.g. ` decaffeinate: Rename AbTestingManager.coffee and 189 other files from .coffee to .js` and then `decaffeinate: Convert AbTestingManager.coffee and 189 other files to JS`.

I think if we wanted to decaffeinate first, we would have to do some odd renaming to keep the git history.
```bash
decaffeinate app.coffee

git mv app.js app.coffee

git commit -m "..."

git mv app.coffee app.js
```

To me, both methods are a little confusing, but this one is a little bit weird IMO. If it's better for you I'm happy to go with that.

Reply:
Ah right that makes sense. In that case happy to keep it as it is then